saplings/circuits/src/components/forms/propose_circuit/service.js

```diff
@@ -14,565 +14,442 @@
+        <InputRow>
+          {basicInput('service-type', 'Service Type')}
+          {basicInput(
+            'service-id',
+            'Service ID',
+            <Button
+              label={<FontAwesomeIcon icon={faSync} />}
+              className="icon-button"
+              onClick={generateId}
             />
-            <div className="form-error">{serviceState.errors.serviceType}</div>
-          </div>
-        </div>
-        <div className="service-field bg-color-grey">
-          <div className="field-name">Service ID</div>
-          <div className="field-input">
-            <div className="flex-input">
-              <input
-                className="service-input service-id-input"
-                value={serviceState.service.serviceId}
-                onChange={e => {
-                  serviceDispatcher({
-                    type: 'set-service-id',
-                    serviceId: e.target.value
-                  });
-                }}
-                maxLength="4"
-                disabled={!editMode}
-              />
-              {editMode && (
-                <button
-                  className="icon-btn"
-                  type="button"
-                  onClick={() => {
-                    const id = generateServiceID();
-                    serviceDispatcher({
-                      type: 'set-service-id',
-                      serviceId: id
-                    });
-                  }}
-                  title="Generate service ID"
-                >
-                  <FontAwesomeIcon icon="sync-alt" />
-                </button>
-              )}
-            </div>
-            <div className="form-error">{serviceState.errors.serviceId}</div>
-          </div>
-        </div>
-        <div className="service-field">
-          <div className="field-name">Allowed node</div>
-          <div className="field-input">
-            {!editMode && allowedNodes}
-            {editMode && allowedNodeEditMode}
-            <div className="form-error">{serviceState.errors.allowedNodes}</div>
-          </div>
-        </div>
-        <div className="service-field bg-color-grey">
-          <div className="field-name">Service arguments</div>
-          <div className="field-input service-arguments">
-            {serviceArguments()}
-          </div>
+          )}
+        </InputRow>
+        <InputRow>
+          <ListBoxSelect
+            name="service-node"
+            label="Node"
+            onChange={onChange}
+            value={values['service-node']}
+            options={[
+              {
+                value: '',
+                content: <div className="no-node-selected">Select a node</div>
+              },
+              ...nodes.map(node => ({
+                value: node.identity,
+                content: <NodeDisplay node={node} />
+              }))
+            ]}
+          />
+        </InputRow>
+        {values['service-args'].map((_, idx) => serviceArgRow(idx))}
```

Comment:
It isn't really clear in the UI what these key value pairs are for

Reply:
Would some place holder text and a section header help?

Reply:
Added a section header